### PR TITLE
Feature/selfdestruct

### PR DESCRIPTION
**Pull request by Aboudjem**
Added new contract Destroyable, with selfDestruct method

### CODE REVIEW

File: contracts/destroyable/Destroyable.sol [hunk 1]
```diff
@@ -0,0 +1,9 @@
+pragma solidity ^0.6.0;
+
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+
+contract Destroyable is Ownable {
+    function kill() public onlyOwner {
```
Comment (Nakasar) [inline contracts/destroyable/Destroyable.sol]: We probably don't need to have this method if it is that simple. We might be able to use a delegatecall to a contract that implements a selfdestruct call.

**Comment by Myna65** [comment]
As explained, all the registries need to be destructed but the contract should be kept and should revert all transactions.

File: flat.sol [hunk 2]
```diff
@@ -0,0 +1,1597 @@
+
```
Comment (Nakasar) [inline flat.sol]: What is this file? Shouldn't it be auto-generated?

**Comment by Myna65** [comment]
Please squash all the commits :dog2:

File: contracts/destroyable/Destroyable.sol [hunk 3]
```diff
@@ -0,0 +1,20 @@
+pragma solidity ^0.6.0;
+
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+
+contract Destroyable is Ownable {
+    bool private authorization;
```
Comment (Myna65) [inline contracts/destroyable/Destroyable.sol]: Please make the name clearer, authorization can mean anything

File: contracts/roles/AgentRole.sol [hunk 4]
```diff
@@ -37,4 +37,5 @@ contract AgentRole is Ownable {
         _agents.remove(account);
         emit AgentRemoved(account);
     }
+
```
Comment (Myna65) [inline contracts/roles/AgentRole.sol]: ?

File: contracts/destroyable/Destroyable.sol (hunk 3 above)
Comment (Aboudjem) [reply contracts/destroyable/Destroyable.sol]: we choose it together....

File: contracts/registry/ClaimTopicsRegistry.sol [hunk 5]
```diff
@@ -2,6 +2,7 @@ pragma solidity ^0.6.0;
 
 import "../registry/IClaimTopicsRegistry.sol";
 import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "../destroyable/Destroyable.sol";
 
 contract ClaimTopicsRegistry is IClaimTopicsRegistry, Ownable {
```
Comment (Joachim-Lebrun) [inline contracts/registry/ClaimTopicsRegistry.sol]:
```suggestion
contract ClaimTopicsRegistry is IClaimTopicsRegistry, Ownable, Destroyable {
```
@dilipror is right, you should add this

Comment (Aboudjem) [reply contracts/registry/ClaimTopicsRegistry.sol]: Yes, forgot this ! Thanks :)

File: contracts/registry/TrustedIssuersRegistry.sol [hunk 6]
```diff
@@ -1,11 +1,11 @@
 pragma solidity ^0.6.0;
 
-
+import "../destroyable/Destroyable.sol";
 import "../registry/ITrustedIssuersRegistry.sol";
 import "@onchain-id/solidity/contracts/IClaimIssuer.sol";
 import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
 
-contract TrustedIssuersRegistry is ITrustedIssuersRegistry, Ownable {
+contract TrustedIssuersRegistry is ITrustedIssuersRegistry, Ownable, Destroyable {
```
Comment (dilipror) [inline contracts/registry/TrustedIssuersRegistry.sol]: Inheriting ownable is not required in trusted issuer registry if we are already inheriting it in Destroyable.

File: contracts/registry/IdentityRegistry.sol [hunk 7]
```diff
@@ -3,14 +3,15 @@ pragma solidity ^0.6.0;
...
 import "@onchain-id/solidity/contracts/IClaimIssuer.sol";
 import "@onchain-id/solidity/contracts/IIdentity.sol";
-import "../registry/IClaimTopicsRegistry.sol";
 import "../registry/ITrustedIssuersRegistry.sol";
+import "../registry/IClaimTopicsRegistry.sol";
 import "../registry/IIdentityRegistry.sol";
+import "../destroyable/Destroyable.sol";
 import "../roles/AgentRole.sol";
 
 import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
```
Comment (dilipror) [inline contracts/registry/IdentityRegistry.sol]: Ownable import should be removed

File: contracts/registry/ClaimTopicsRegistry.sol (hunk 5 above)
Comment (dilipror) [inline contracts/registry/ClaimTopicsRegistry.sol]: Inheriting ownable is not required in claim topics registry if we are already inheriting it in Destroyable.

Comment (Joachim-Lebrun) [reply contracts/registry/ClaimTopicsRegistry.sol]: indeed

**Comment by Joachim-Lebrun** [comment]
there is no tests on the functions created here, can you please check to write the tests @Aboudjem ?

**Comment by Joachim-Lebrun** [comment]
i think you should restart from scratch on the beta branch or it will be very complicated to merge

**Pull request closed without merging**